app/code/Magento/Review/Test/Unit/Observer/CatalogProductListCollectionAppendSummaryFieldsObserverTest.php

```diff
@@ -0,0 +1,138 @@
+use Magento\Framework\Event;
+use Magento\Framework\Event\Observer;
+use Magento\Review\Model\ResourceModel\Review\Summary;
+use Magento\Review\Model\ResourceModel\Review\SummaryFactory;
+use Magento\Review\Observer\CatalogProductListCollectionAppendSummaryFieldsObserver;
+use Magento\Store\Api\Data\StoreInterface;
+use Magento\Store\Model\StoreManagerInterface;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Test class for \Magento\Review\Observer\CatalogProductListCollectionAppendSummaryFieldsObserver
+ */
+class CatalogProductListCollectionAppendSummaryFieldsObserverTest extends TestCase
+{
+    /**
+     * @var Event|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $eventMock;
+
+    /**
+     * Testable Object
+     *
+     * @var CatalogProductListCollectionAppendSummaryFieldsObserver
+     */
+    private $observer;
+
+    /**
+     * @var Observer|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $observerMock;
+
+    /**
+     * @var Collection|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $productCollectionMock;
+
+    /**
+     * @var StoreInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $storeMock;
+
+    /**
+     * @var StoreManagerInterface|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $storeManagerMock;
+
+    /**
+     * @var Summary|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $sumResourceMock;
+
+    /**
+     * @var SummaryFactory|\PHPUnit_Framework_MockObject_MockObject
+     */
+    private $sumResourceFactoryMock;
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp() : void
+    {
+        $this->eventMock = $this->getMockBuilder(Event::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['getCollection'])
+            ->getMock();
+
+        $this->observerMock = $this->createMock(Observer::class);
+
+        $this->productCollectionMock = $this->getMockBuilder(Collection::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->storeManagerMock = $this->getMockBuilder(StoreManagerInterface::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['getStore'])
+            ->getMockForAbstractClass();
+
+        $this->storeMock = $this->getMockBuilder(StoreInterface::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['getId'])
+            ->getMockForAbstractClass();
+
+        $this->sumResourceFactoryMock = $this->getMockBuilder(SummaryFactory::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['create'])
+            ->getMock();
+
+        $this->sumResourceMock = $this->getMockBuilder(Summary::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['appendSummaryFieldsToCollection'])
+            ->getMock();
+
+        $this->observer = new CatalogProductListCollectionAppendSummaryFieldsObserver(
```

Comment:
Use ObjectManager helper: `\Magento\Framework\TestFramework\Unit\Helper\ObjectManager`

Reply:
**done**

Reply:
OK. I'll investigate that further later on. But for now, it's fine for me :+1: